app/helpers/social_accounts_helper.rb

```diff
@@ -0,0 +1,30 @@
+module SocialAccountsHelper
+  def generate_link(social_account)
+    network = social_account.social_network
+      if network == "GitHub"
+        return "https://github.com/#{social_account.user_name}"
+      end
+      if network == "Telegram"
+        return "https://t.me/#{social_account.user_name}"
+      end
+      if network == "Facebook"
+        return "https://www.facebook.com/#{social_account.user_name}"
+      end
+      if network == "Twitter"
+        return "https://twitter.com/#{social_account.user_name}"
+      end
+      if network == "GitLab@HPI"
+        return "https://twitter.com/#{social_account.user_name}"
+      end
+      if network == "Slack"
+        return "https://slack.com/"
+      end
+      if network == "Discord"
+        return "https://discordapp.com/#{social_account.user_name}"
+      end
+      return "#"
+  end
+  def get_supported_social_networks 
+    return [["GitHub", "GitHub"], ["Telegram", "Telegram"], ["Facebook", "Facebook"], ["Twitter", "Twitter"], ["GitLab@HPI", "GitLab@HPI"], ["Slack", "Slack"], ["Discord", "Discord"]]
```

Comment:
The names of the social networks are in all languages the same 🙃 . So I think we do not need to translate them.

Reply:
I think that is only true if you already know which languages should be supported in the future. At openHPI we support Chinese and Russian and they do not use the latin alphabet so maybe (I do not know) services could have another name in countries like these. It would then be a lot of work to find all the things previously considered as language independent so in my opinion all strings displayed to a user should be extracted to the locale files.